role/defenders.py

```diff
@@ -0,0 +1,251 @@
+	def execute_setup(self):
+		pass
+
+	def on_exit_setup(self):
+		pass
+
+	def on_enter_block(self):
+		pass
+
+	def execute_block(self):
+		ball=Vector2D(self.kub.state.ballPos)
+		goal_line=Line(point1=Vector2D(self.kub.state.ballPos),point2=Vector2D(-HALF_FIELD_MAXX,0))
+
+		target_point=self.line.intersection_with_line(goal_line)
+
+		if target_point.y > 0.5*DBOX_WIDTH+5*BOT_RADIUS:
+			target_point=self.line2.intersection_with_line(goal_line)
+			if self.def_type=="top":
+				target_point.x-=BOT_RADIUS
+			else:
+				target_point.x+=BOT_RADIUS
+
+		elif target_point.y < -0.5*DBOX_WIDTH-5*BOT_RADIUS:
+			target_point=self.line1.intersection_with_line(goal_line)
+			if self.def_type=="top":
+				target_point.x+=BOT_RADIUS
+			else:
+				target_point.x-=BOT_RADIUS
+
+		else:
+			if self.def_type=="top":
+				target_point.y+=BOT_RADIUS
+			else:
+				target_point.y-=BOT_RADIUS
+
+		if ball_moving_towards_our_goal(self.kub.state):
+			ball_line=Line(point1=Vector2D(self.kub.state.ballPos),angle=math.atan2(self.kub.state.ballVel.y,self.kub.state.ballVel.x))
+			point=ball_line.intersection_with_line(self.goal_line)
+			if point.y*point.y<0.35*DBOX_WIDTH*DBOX_WIDTH :
+				target_point=self.line.intersection_with_line(ball_line)
+				if target_point.y > 0.5*DBOX_WIDTH+5*BOT_RADIUS:
+					target_point=self.line2.intersection_with_line(ball_line)
+					if self.def_type=="top":
+						target_point.x-=BOT_RADIUS
+					else:
+						target_point.x+=BOT_RADIUS	
+				elif target_point.y < -0.5*DBOX_WIDTH-5*BOT_RADIUS:
+					target_point=self.line1.intersection_with_line(ball_line)
+					if self.def_type=="top":
+						target_point.x+=BOT_RADIUS
+					else:
+						target_point.x-=BOT_RADIUS
+				else:
+					if self.def_type=="top":
+						target_point.y+=BOT_RADIUS
+					else:
+						target_point.y-=BOT_RADIUS
+
+		_GoToPoint_.init(self.kub, target_point, 0)
+		start_time = rospy.Time.now()
+		start_time = 1.0*start_time.secs + 1.0*start_time.nsecs/pow(10,9)   
+		generatingfunction = _GoToPoint_.execute(start_time,self.course_approch_thresh,True)
+		
+		prev_point=Vector2D(self.kub.state.ballPos)
+		next_point = prev_point
+		movement=self.kub.get_pos()
+		point=Vector2D()
+		for gf in generatingfunction:
+			next_point = Vector2D(self.kub.state.ballPos)
+			self.kub,point = gf
+			if dist(next_point,prev_point) > 3*BOT_RADIUS:
+				break
+			if not vicinity_points(point,target_point,thresh=BOT_RADIUS*2.0):
+				break
+			if self.should_intercept():
+				break
+			if self.ready_to_kick():
+				break 	
+			prev_point=next_point
+
+	def on_exit_block(self):
+		pass
+
+	def intercept_complete(self):
+		ball_vel = Vector2D(self.kub.state.ballVel.y,self.kub.state.ballVel.x)
+		ball_vel_angle = ball_vel.tan_inverse()
+		bot_ball_dir = Vector2D(self.kub.state.ballPos.y-self.kub.state.homePos[self.kub.kubs_id].y , self.kub.state.ballPos.x-self.kub.state.homePos[self.kub.kubs_id].x)
+		if ( abs(ball_vel_angle - bot_ball_dir.tan_inverse() )< 0.0523599):
+			return 1
```

Comment:
Avoid using random numbers.

Reply:
@nisarg1631 Instead of using Angle threshold like this, make a variable in config file and use that variable here. Once you do that, I think we can merge this.